lib/open_api/docs/doc_v2.rb

```diff
@@ -39,6 +51,44 @@ def routes
           end
         end
       end
+
+      private
+
+      def substitute_references(object)
+        if object.kind_of?(Array)
+          object.collect { |i| substitute_references(i) }
+        elsif object.kind_of?(Hash)
+          return fetch_ref_value(object["$ref"]) if object.keys == ["$ref"]
+          object.each_with_object({}) do |(k, v), h|
+            h[k] = substitute_references(v)
+          end
```

Comment:
I don't think you need to create a new Hash here...can this work?

```ruby
          return fetch_ref_value(object["$ref"]) if object.keys == ["$ref"]
          object.keys.each do |k|
            object[k] = substitute_references(object[k])
          end
```

Reply:
Note I'm doing .keys.each here to avoid modifying a hash while it's being iterated over